src/category_theory/monoidal/of_has_finite_products.lean

```diff
@@ -16,48 +16,92 @@ A category with a monoidal structure provided in this way is sometimes called a
 although this is also sometimes used to mean a finitely complete category.
 (See <https://ncatlab.org/nlab/show/cartesian+category>.)
 
-As this works with either products or coproducts, we don't set up either construct as an instance.
--/
+As this works with either products or coproducts,
+and sometimes we want to think of a different monoidal structure entirely,
+we don't set up either construct as an instance.
 
-open category_theory.limits
+## Implementation
+For the sake of nicer definitional properties,
+we rely on `has_terminal` and `has_binary_products` instead of `has_finite_products`,
+so that if a particular category provides customised instances of these
+we pick those up instead.
+-/
 
 universes v u
 
 namespace category_theory
+open category_theory.limits
 
-section
 variables (C : Type u) [𝒞 : category.{v} C]
 include 𝒞
 
+section
 local attribute [tidy] tactic.case_bash
 
-/-- A category with finite products has a natural monoidal structure. -/
-def monoidal_of_has_finite_products [has_finite_products.{v} C] : monoidal_category C :=
-{ tensor_unit  := terminal C,
-  tensor_obj   := λ X Y, limits.prod X Y,
+/-- A category with a terminal object and binary products has a natural monoidal structure. -/
+def monoidal_of_has_finite_products [has_terminal.{v} C] [has_binary_products.{v} C] : monoidal_category C :=
```

Comment:
Should we also change the name?

Reply:
Yeah, I was indecisive. It's maths-same, and the new name would be a mouthful.

Reply:
Yep, I agree. I'm fine with leaving it like this.